### PR TITLE
Sans JSON Serialization

### PR DESCRIPTION
**Pull request by stas**
First steps to completely remove the out-of-the-box JSON serialization support.

I added an example how to achieve the same old API functionality.

This should close #12 

Thanks to @ThomasSevestre  and @christophersansone for helping with this!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -126,7 +126,7 @@ hash = MovieSerializer.new(movie).serializable_hash
 
 #### Return Serialized JSON
 ```ruby
-json_string = MovieSerializer.new(movie).serialized_json
+json_string = MovieSerializer.new(movie).serialized_hash.to_json
```
Comment (doits) [inline README.md]: Is it possible that this should read `serializable_hash`?:

```suggestion
json_string = MovieSerializer.new(movie).serializable_hash.to_json
```

File: README.md [hunk 2]
```diff
@@ -324,7 +324,7 @@ options[:links] = {
   prev: '...'
 }
 options[:include] = [:actors, :'actors.agency', :'actors.agency.state']
-MovieSerializer.new(movies, options).serialized_json
+MovieSerializer.new(movies, options).serialized_hash
```
Comment (doits) [inline README.md]: ... here too with a `.to_json`?

```suggestion
MovieSerializer.new(movies, options).serializable_hash.to_json
```

File: README.md [hunk 3]
```diff
@@ -337,7 +337,7 @@ options[:links] = {
   prev: '...'
 }
 hash = MovieSerializer.new(movies, options).serializable_hash
-json_string = MovieSerializer.new(movies, options).serialized_json
+json_string = MovieSerializer.new(movies, options).serialized_hash
```
Comment (doits) [inline README.md]: ... and here? :-)

```suggestion
json_string = MovieSerializer.new(movies, options).serializable_hash.to_json
```

... or even ...?

```suggestion
json_string = hash.to_json
```

**Comment by doits** [comment]
Just my two cents: :+1: for removing it, I think that this should not be part of the library either.

**Comment by stas** [comment]
Thanks @doits, I forgot to update the documentation, will do!

**Comment by oyeanuj** [comment]
@stas Is this good to be merged?

**Comment by stas** [comment]
@oyeanuj yep, I should merge it, was trying to orchestrate it a bit together with other updates...

**Comment by stas** [comment]
Merged into the wrong branch, so I updated it manually... :see_no_evil:

File: README.md [hunk 4]
```diff
@@ -126,7 +126,7 @@ hash = MovieSerializer.new(movie).serializable_hash
 
 #### Return Serialized JSON
 ```ruby
-json_string = MovieSerializer.new(movie).serialized_json
+json_string = MovieSerializer.new(movie).serializable_hash_hash.to_json
```
Comment (nicbet) [inline README.md:129]: extra `_hash`, should be `serializable_hash.to_json`

File: docs/json_serialization.md [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+require 'oj'
+require 'fast_jsonapi'
+
+class BaseSerializer
+  include FastJsonapi::ObjectSerializer
+
+  def to_json
+    Oj.dump(serializable_hash)
+  end
+  alias_method :serialized_json :to_json
```
Comment (nicbet) [inline docs/json_serialization.md:21]: missing `,` - should be `alias method :serialized_json, :to_json`

File: docs/json_serialization.md [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+it working with the popular `oj` gem:
+
+```ruby
+require 'oj'
+require 'fast_jsonapi'
+
+class BaseSerializer
+  include FastJsonapi::ObjectSerializer
+
+  def to_json
```
Comment (nicbet) [inline docs/json_serialization.md:18]: leads to `ArgumentError (wrong number of arguments (given 1, expected 0))`

**Comment by nicbet** [comment]
Was hit by the Deprecation warning today, tried to follow changes in README and hit some issues. I left comments on the PR commit code where things are amiss.

**Comment by stas** [comment]
Thanks @nicbet, but this is long ago fixed on master. In future, could you please open a separate issue, it's a bit hard to follow the conversations after the PRs are merged. :bow:
